diff --git a/node_modules/ai/dist/index.mjs b/node_modules/ai/dist/index.mjs
index f3e132c..832c55c 100644
--- a/node_modules/ai/dist/index.mjs
+++ b/node_modules/ai/dist/index.mjs
@@ -424,7 +424,7 @@ function formatWarning({
 var FIRST_WARNING_INFO_MESSAGE = "AI SDK Warning System: To turn off warning logging, set the AI_SDK_LOG_WARNINGS global to false.";
 var hasLoggedBefore = false;
 var logWarnings = (options) => {
-  if (options.warnings.length === 0) {
+  if (!options || !options.warnings || options.warnings.length === 0) {
     return;
   }
   const logger = globalThis.AI_SDK_LOG_WARNINGS;
@@ -8287,7 +8287,9 @@ async function embedMany({
         );
         for (const result of results) {
           embeddings.push(...result.embeddings);
-          warnings.push(...result.warnings);
+          if (result.warnings) {
+            warnings.push(...result.warnings);
+          }
           responses.push(result.response);
           tokens += result.usage.tokens;
           if (result.providerMetadata) {
